ext/opentelemetry-ext-datadog/tests/test_datadog_exporter.py

```diff
@@ -0,0 +1,267 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import unittest
+from unittest import mock
+
+from ddtrace.internal.writer import AgentWriter
+
+# pylint:disable=no-name-in-module
+# pylint:disable=import-error
+import opentelemetry.ext.datadog as datadog_exporter
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk import trace
+from opentelemetry.sdk.trace import export
+
+
+class TestDatadogSpanExporter(unittest.TestCase):
+    def setUp(self):
+        tracer_provider = trace.TracerProvider()
+
+        self.exporter = datadog_exporter.DatadogSpanExporter()
+
+        # just agent is configured now
+        self.agent_writer_mock = mock.Mock(spec=AgentWriter)
+        self.agent_writer_mock.started = True
+        self.agent_writer_mock.exit_timeout = 1
+
+        # pylint: disable=protected-access
+        self.exporter._agent_writer = self.agent_writer_mock
+
+        tracer_provider.add_span_processor(
+            export.SimpleExportSpanProcessor(self.exporter)
+        )
+        self.tracer = tracer_provider.get_tracer(__name__)
+
+    def get_spans(self):
+        kwargs_spans = [
+            call_args[-1]["spans"]
+            for call_args in self.agent_writer_mock.write.call_args_list
+        ]
+        return [
+            span.to_dict()
+            for span in itertools.chain.from_iterable(kwargs_spans)
+        ]
+
+    def test_constructor_default(self):
+        """Test the default values assigned by constructor."""
+        exporter = datadog_exporter.DatadogSpanExporter()
+
+        self.assertEqual(exporter.agent_url, "http://localhost:8126")
+        self.assertTrue(exporter.service is None)
```

Comment:
```suggestion
        self.assertIsNone(exporter.service)
```

Similar ones in other places.